syntax = "proto3";

package syft.core.node.common.action;

import "proto/core/common/common_object.proto";
import "proto/core/pointer/pointer.proto";
import "proto/core/io/address.proto";

message RunClassMethodSMPCAction {
  string path = 1;
  bytes _self = 2;
  repeated syft.core.pointer.Pointer args = 3;
  map<string, syft.core.pointer.Pointer> kwargs = 4;
  syft.core.common.UID id_at_location = 5;
  string seed_id_locations = 6;
  syft.core.io.Address address = 7;
  syft.core.common.UID msg_id = 8;
}
